src/prng/mod.rs

```diff
@@ -59,82 +59,72 @@
+//! performance bottleneck.
 //!
-//! When choosing a PRNG for performance it is very important to benchmark your
-//! own application due to interactions between PRNGs and surrounding code and
+//! Performance of normal PRNGs is a bit of a subtle thing. It depends a lot on
+//! the CPU architecture (32 vs. 64 bits), inlining, and also on the number of
+//! available registers. This often causes the performance to be affected by
+//! surrounding code due to inlining and other usage of registers.
+//!
+//! When choosing a PRNG for performance it is important to benchmark your own
+//! application due to interactions between PRNGs and surrounding code and
 //! dependence on the CPU architecture as well as the impact of the size of
 //! data requested. Because of all this, we do not include performance numbers
-//! here but merely a qualitative description.
+//! here but merely a qualitative rating (which is not comparable between PRNGs
+//! and CSPRNGs).
 //!
 //! CSPRNGs are a little different in that they typically generate a block of
 //! output in a cache, and pull outputs from the cache. This allows them to have
 //! good amortised performance, and reduces or completely removes the influence
-//! of surrounding code on the CSPRNG performance.
-//!
-//! ## Worst-case performance
-//!
-//! Simple PRNGs tend to use constant-time functions and thus have very good
-//! worst case performance. In contrast, CSPRNGs usually produce a block of
-//! values into a cache, giving them poor worst case performance.
+//! of surrounding code on the CSPRNG performance. It does however cause *poor
+//! worst case performance*.
 //!
 //! ## State size
 //!
-//! Simple PRNGs often use very little memory, from as little as one word but
-//! commonly 2-4 words, where a *word* is usually either `u32` or `u64`. This
-//! is not universal across non-cryptographic PRNGs however; for example the
-//! previously popular Mersenne Twister MT19937 algorithm requires 2.5 kB of
-//! state.
+//! Normal PRNGs often use very little memory, commonly only a few words, where
+//! a *word* is usually either `u32` or `u64`. This is not universal however,
+//! for example the historically popular Mersenne Twister MT19937 algorithm
+//! requires 2.5 kB of state.
 //!
 //! CSPRNGs typically require more memory; since the seed size is recommended
-//! to be at least 192 bits and more memory is required for a counter, 256 bits
-//! would be approximately the minimum secure size. In practice, CSPRNGs tend
-//! to use a lot of memory; [`ChaChaRng`] is relatively small with 512 bits
-//! state + 512 bits cache + cache counter (136 bytes on x86_64) while [`Hc128Rng`]
-//! requires over 4 kB.
+//! to be at least 192 bits and some more may be required for the algorithm,
+//! 256 bits would be approximately the minimum secure size. In practice,
+//! CSPRNGs tend to use quite a bit more, [`ChaChaRng`] is relatively small with
+//! 136 bytes.
```

Comment:
Don't want to be a grammar nazi but "with 136 bytes" should be followed by "of ..." or something; also better to use `more;` than `more,` here.

Reply:
I can take a little well-meant critique on my writing :smile: